src/tls_crypt.ml

```diff
@@ -49,6 +55,32 @@ let pem_of_lines ~name seq =
         Base64.decode b64
   | _ -> error_msgf "Invalid %s" name
 
+let hex_of_lines ~name seq =
+  let seq =
+    Seq.drop_while (fun str -> String.length str > 0 && str.[0] = '#') seq
+  in
+  match List.of_seq seq with
+  | [ _ ] -> error_msgf "Invalid %s" name
+  | header :: hex_and_footer when header = "-----BEGIN " ^ name ^ "-----" -> (
+      let hex_and_footer = List.trim hex_and_footer in
+      let footer = List.hd (List.rev hex_and_footer) in
+      if footer <> "-----END " ^ name ^ "-----" then
+        error_msgf "Invalid %s" name
+      else
+        let hex = List.chop hex_and_footer in
+        let buf = Buffer.create 0x100 in
+        let f str =
+          for i = 0 to (String.length str / 2) - 1 do
```

Comment:
If the number of characters in a line is odd the last character will be ignored. Lucky for us this means carriage returns will **usually** be ignored. However, if a line ends with a space **and** a carriage return this will cause trouble.

The OpenVPN implementation seems to ignore all whitespace. However, it seems they ignore the last hex digit if there is an odd number of total hex digits(!)
https://github.com/OpenVPN/openvpn/blob/f015643fe23d7847ad45b7763f31bfc6baed2159/src/openvpn/crypto.c#L1257-L1289